nifi-commons/nifi-record-path/src/test/java/org/apache/nifi/record/path/TestRecordPath.java

```diff
@@ -1008,12 +1008,16 @@ public void testReplaceRegex() {
         final List<RecordField> fields = new ArrayList<>();
         fields.add(new RecordField("id", RecordFieldType.INT.getDataType()));
         fields.add(new RecordField("name", RecordFieldType.STRING.getDataType()));
```

Comment:
There should be tests for all the escapable chars

Reply:
added tests to cover all of special chars. Also added backslash escape sequences to test exceptions.